File: samples/ble_util/src/com/idevicesinc/sweetblue/ble_util/BleStatusBar.java
package com.idevicesinc.sweetblue.ble_util;

import android.content.Context;
import android.text.SpannableString;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.Button;
import android.widget.CompoundButton;
import android.widget.CompoundButton.OnCheckedChangeListener;
import android.widget.FrameLayout;
import android.widget.TextView;
import android.widget.ToggleButton;

import com.idevicesinc.sweetblue.*;
import com.idevicesinc.sweetblue.utils.Utils;

/**
 * 
 * @author dougkoellmer
 */
public class BleStatusBar extends FrameLayout implements BleManager.StateListener, BleManager.NativeStateListener
{
	private final BleManager m_bleMngr;
	
	private final TextView m_status;
	private final TextView m_nativeStatus;
	
	public BleStatusBar(Context context, BleManager bleMngr)
	{
		super(context);
		
		m_bleMngr = bleMngr;
		m_bleMngr.setListener_State(this);
		m_bleMngr.setListener_NativeState(this);
		
		LayoutInflater li = LayoutInflater.from(context);
		View inner = li.inflate(R.layout.ble_status_bar, null);
		
		m_status = (TextView) inner.findViewById(R.id.ble_status);
		m_nativeStatus = (TextView) inner.findViewById(R.id.native_ble_status);
		
		updateStatus();
		
		this.addView(inner);
	}
	
	private void updateStatus()
	{
		SpannableString status = Utils.makeStateString(BleState.values(), m_bleMngr.getStateMask());
		m_status.setText(status);
		
		status = Utils.makeStateString(BleState.values(), m_bleMngr.getNativeStateMask());
		m_nativeStatus.setText(status);
	}
	
	@Override public void onBleStateChange(BleManager manager, int oldStateBits, int newStateBits)
	{
		updateStatus();
	}

	@Override public void onNativeBleStateChange(BleManager manager, int oldStateBits, int newStateBits)
	{
		updateStatus();
	}
}

File: xamarin/SweetBlue/component/samples/Android/BleUtil/BleUtil/BleStatusBar.cs
using System;
using Android.App;
using Android.Text;
using Android.Widget;
using Android.Views;
using Android.Content;
using Idevices.Sweetblue;
using Idevices.Sweetblue.Util;


namespace BleUtil
{
	public class BleStatusBar : FrameLayout
	{

		private BleManager m_bleMgr;
		private TextView m_status;
		private TextView m_nativeStatus;


		public BleStatusBar (Context context, BleManager mgr) : base(context)
		{
			m_bleMgr = mgr;
			m_bleMgr.State += delegate {
				updateStatus();		
			};
			m_bleMgr.NativeState += delegate {
				updateStatus();	
			};

			View inner = LayoutInflater.From (context).Inflate (Resource.Layout.Ble_Status_Bar, null);

			m_status = inner.FindViewById<TextView> (Resource.Id.ble_status);
			m_nativeStatus = inner.FindViewById<TextView> (Resource.Id.native_ble_status);

			updateStatus ();

			AddView (inner);
		}

		private void updateStatus()
		{
			SpannableString status = Utils_String.MakeStateString (BleManagerState.Values(), m_bleMgr.StateMask);
			m_status.TextFormatted = status;

			status = Utils_String.MakeStateString (BleManagerState.Values (), m_bleMgr.NativeStateMask);
			m_nativeStatus.TextFormatted = status;
		}

	}
}

